en/tutorials-and-examples/cms/articles-controller.rst

```diff
@@ -247,11 +249,289 @@ to :php:func:`Cake\\Routing\\Router::url()` function on the `API
+validation engine is powerful and flexible. It provides a suite of frequently
+used rules for tasks like email addresses, IP addresses etc. and flexibility for
+adding your own validation rules. For more information on that setup, check the
+:doc:`/core-libraries/validation` documentation.
+
+Now that your validation rules are in place, use the app to try to add
+an article with an empty title or body to see how it works.  Since we've used the
+:php:meth:`Cake\\View\\Helper\\FormHelper::control()` method of the FormHelper to
+create our form elements, our validation error messages will be shown
+automatically.
+
+Add Delete Action
+=================
+
+Next, let's make a way for users to delete articles. Start with a
+``delete()`` action in the ``ArticlesController``::
+
+    // src/Controller/ArticlesController.php
+
+    public function delete($slug)
+    {
+        $this->request->allowMethod(['post', 'delete']);
+
+        $article = $this->Articles->findBySlug($slug)->firstOrFail();
+        if ($this->Articles->delete($article)) {
+            $this->Flash->success(__('The {0} article has been deleted.', $article->title));
+            return $this->redirect(['action' => 'index']);
+        }
+    }
+
+This logic deletes the article specified by ``$slug``, and uses
+``$this->Flash->success()`` to show the user a confirmation
+message after redirecting them to ``/articles``. If the user attempts to
+delete an article using a GET request, ``allowMethod()`` will throw an Exception.
+Uncaught exceptions are captured by CakePHP's exception handler, and a nice
+error page is displayed. There are many built-in
+:doc:`Exceptions </development/errors>` that can be used to indicate the various
+HTTP errors your application might need to generate.
+
+.. warning::
+
+    Allowing content to be deleted using GET requests is dangerous, as web
+    crawlers could accidentally delete all your content. That is why we used
+    ``allowMethod()`` in our controller.
+
+Because we're just executing some logic and redirecting, this action has no
+view. You might want to update your index view with links that allow users to
+delete articles:
+
+.. code-block:: php
+
+    <!-- File: src/Template/Articles/index.ctp  (deleted links added) -->
+
+    <h1>Articles</h1>
+    <p><?= $this->Html->link("Add Article", ['action' => 'add']) ?></p>
+    <table>
+        <tr>
+            <th>Title</th>
+            <th>Created</th>
+            <th>Action</th>
+        </tr>
+
+    <!-- Here's where we iterate through our $articles query object, printing out article info -->
+
+    <?php foreach ($articles as $article): ?>
+        <tr>
+            <td>
+                <?= $this->Html->link($article->title, ['action' => 'view', $article->slug]) ?>
+            </td>
+            <td>
+                <?= $article->created->format(DATE_RFC850) ?>
+            </td>
+            <td>
+                <?= $this->Html->link('Edit', ['action' => 'edit', $article->slug]) ?>
```

Comment:
Edit uses `$article->id` still. I wonder if it should be changed throughout the tutorial to use `$slug` the way all the other methods use it.